src/main/java/io/github/syst3ms/skriptparser/util/math/NumberMath.java

```diff
@@ -200,6 +201,50 @@ public static Number tanh(Number n) {
         }
     }
 
+    /**
+     * Checks if a given BigInteger is prime. The certainty of this check is always 100%
+     * @param number the number to check
+     * @return whether or not this number is a prime
+     */
+    public static boolean isPrime(BigInteger number) {
+        // 1. Check if the number is of the form 6x±1 for values higher than 3.
+        if (!number.subtract(BigInteger.ONE).mod(BigInteger.valueOf(6)).equals(BigInteger.ZERO)
+                && !number.add(BigInteger.ONE).mod(BigInteger.valueOf(6)).equals(BigInteger.ZERO)
+                && number.compareTo(BigInteger.valueOf(3)) > 0)
+            return false;
+
+        // 2. Check in the cached primes.
+        if (cachedPrimes.contains(number.intValue()))
```

Comment:
I'm sorry for being so particular about this, but since cachedPrimes is known to be sorted, this can be significantly sped up by using Collections#binarySearch as such:
```suggestion
        if (Collections.binarySearch(cachedPrimes, number.intValue()) >= 0)
```
This isn't the most relevant thing, but it can be helpful to know: note that a `!= -1` won't cut it in this case, since if the index isn't found, binarySearch() returns _negative_ the index at which the tested element would have to be inserted for the list to remain sorted, minus one (otherwise this causes a problem if the element belongs at the start of the list).